subcommand/inject-connect/command.go

```diff
@@ -57,17 +80,35 @@ func (c *Command) init() {
 	c.flagSet.StringVar(&c.flagKeyFile, "tls-key-file", "",
 		"PEM-encoded TLS private key to serve. If blank, will generate random cert.")
 	c.flagSet.StringVar(&c.flagConsulImage, "consul-image", connectinject.DefaultConsulImage,
-		"Docker image for Consul. Defaults to an Consul 1.3.0.")
+		"Docker image for Consul. Defaults to an Consul 1.5.0.")
 	c.flagSet.StringVar(&c.flagEnvoyImage, "envoy-image", connectinject.DefaultEnvoyImage,
-		"Docker image for Envoy. Defaults to Envoy 1.8.0.")
+		"Docker image for Envoy. Defaults to Envoy 1.9.1.")
 	c.flagSet.StringVar(&c.flagConsulK8sImage, "consul-k8s-image", "",
 		"Docker image for consul-k8s. Used for the connect sidecar.")
 	c.flagSet.StringVar(&c.flagACLAuthMethod, "acl-auth-method", "",
 		"The name of the Kubernetes Auth Method to use for connectInjection if ACLs are enabled.")
-	c.flagSet.BoolVar(&c.flagCentralConfig, "enable-central-config", false,
+	c.flagSet.BoolVar(&c.flagWriteServiceDefaults, "enable-central-config", false,
 		"Write a service-defaults config for every Connect service using protocol from -default-protocol or Pod annotation.")
 	c.flagSet.StringVar(&c.flagDefaultProtocol, "default-protocol", "",
 		"The default protocol to use in central config registrations.")
+	c.flagSet.BoolVar(&c.flagEnableNamespaces, "enable-namespaces", false,
+		"Enables namespaces, in either a single Consul namespace or mirrored [Enterprise only feature]")
+	c.flagSet.StringVar(&c.flagConsulNamespace, "consul-namespace", "default",
+		"Defines which Consul namespace to register all injected services into. If `enable-namespace-mirroring` "+
+			"is true, this is not used.")
+	c.flagSet.Var((*flags.AppendSliceValue)(&c.flagAllowK8sNamespacesList), "allow-namespace",
+		"K8s namespaces to explicitly allow. May be specified multiple times.")
```

Comment:
Why not do a comma separated list? Might be nicer to use for users

Reply:
We can expose this any way we like in the Helm chart, then do the translation there. This also fits existing patterns, as demonstrated by the server's extra volumes: https://github.com/hashicorp/consul-helm/blob/master/templates/server-statefulset.yaml#L105-L109

Reply:
Okay makes sense. I guess in the end the UX is more about how it's set in the helm chart.